orca-core/src/test/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/ServiceKeyExpressionFunctionProviderTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2019 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.expressions.functions;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.netflix.spinnaker.orca.ExecutionStatus.RUNNING;
+import static com.netflix.spinnaker.orca.ExecutionStatus.SUCCEEDED;
+import static com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.PIPELINE;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ServiceKeyExpressionFunctionProviderTest {
```

Comment:
For future reference, I still believe we can author tests in groovy (we just would prefer non-tests be java/kotlin).

Unit tests in strict java are still a little verbose for my liking (don't need to do this as part of this PR but consider it next time).

Reply:
Is it okay to mix Groovy and Kotlin in the test section of a subproject? One reason that we are avoiding adding Groovy, even in tests, is to clear the way to use more Kotlin in the future and there are apparently issues with mixing Groovy and Kotlin.
Also, there are a lot of ways to write Kotlin tests and none of them are as succinct as Spock. Can you point to ideal examples? We typically use Kotlin, JUnit, assert4j, and mockito, but I've seen variance in the Spinnaker projects.

Reply:
I believe it is fine ... I also suspect we should be more strict on what ends up living in a `-core` artifact.